### PR TITLE
Update font-jetbrains-mono from 1.0.6 to 2.001

### PR DESCRIPTION
**Pull request by MindTooth**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

### CODE REVIEW

**Comment by MindTooth** [comment]
The whole process of release 2.0 is completely redone.

Does it look okay?  No issue installing locally.

**Comment by MindTooth** [comment]
Screen Shot 2020-07-10 at 14 52 43

Does seem that it introduces some duplicated.

**Comment by MTCoster** [comment]
> Does seem that it introduces some duplicated.

This appears to be due to macOS’ handling of OTF variable fonts; fixed-weight versions are automatically generated for the standard named weights (see the advice given [here](https://github.com/Homebrew/homebrew-cask-fonts/pull/2101/files) for details on how `cask-fonts` may handle this)

**Comment by MindTooth** [comment]
Would you recommend that I remove the variable variant?

**Comment by MTCoster** [comment]
@MindTooth I would split it into its own cask, `font-jetbrains-mono-variable`

**Comment by MindTooth** [comment]
I’ll update this tomorrow, and I’ll look into submitting a new cask.

Thanks for the help. 😊

**Comment by MindTooth** [comment]
I've removed the variable variant.  One bug with italic though, so I'll wait on a new release for next week.

Ref. https://github.com/JetBrains/JetBrainsMono/issues/266

**Comment by vitorgalvao** [comment]
> I would split it into its own cask, `font-jetbrains-mono-variable`

Absolutely not. It’s either one or the other. This repo is a big enough mess as it is.

> I've removed the variable variant.

Doing the reverse might be better. But never include both, just the variable or the non-variable.

File: Casks/font-jetbrains-mono.rb [hunk 1]
```diff
@@ -1,27 +1,39 @@
...
-  font "JetBrainsMono-#{version}/ttf/JetBrainsMonoNL-ExtraBold.ttf"
-  font "JetBrainsMono-#{version}/ttf/JetBrainsMono-Italic.ttf"
-  font "JetBrainsMono-#{version}/ttf/JetBrainsMonoNL-Italic.ttf"
-  font "JetBrainsMono-#{version}/ttf/JetBrainsMono-Medium-Italic.ttf"
-  font "JetBrainsMono-#{version}/ttf/JetBrainsMonoNL-Medium-Italic.ttf"
-  font "JetBrainsMono-#{version}/ttf/JetBrainsMono-Medium.ttf"
-  font "JetBrainsMono-#{version}/ttf/JetBrainsMonoNL-Medium.ttf"
-  font "JetBrainsMono-#{version}/ttf/JetBrainsMono-Regular.ttf"
-  font "JetBrainsMono-#{version}/ttf/JetBrainsMonoNL-Regular.ttf"
+  font "JetBrains\ Mono\ #{version}/ttf/JetBrainsMono-Bold-Italic.ttf"
```
Comment (vitorgalvao) [inline Casks/font-jetbrains-mono.rb:11]: Remove the backslashes.

**Comment by MindTooth** [comment]
> Doing the reverse might be better. But never include both, just the variable or the non-variable.

I'll change it if desired?

**Comment by vitorgalvao** [comment]
> I'll change it if desired?

I’m fine either way. Which one has the bug you mentioned above?

**Comment by MindTooth** [comment]
> I’m fine either way. Which one has the bug you mentioned above?

Fixed with version 2.001 released today.

Though it seems that the variable variant is bugged for now, so this is more stable in a sense I guess. 🤔 

Ref.: https://github.com/JetBrains/JetBrainsMono/issues/270

**Comment by vitorgalvao** [comment]
Alright, let’s keep this one. Feel free to switch to the variable version at any time.

Thank you.

**Pull request merged**